datumaro/components/operations.py

```diff
@@ -145,14 +149,44 @@ def _merge_attrs(a: Dict[str, Any], b: Dict[str, Any],
 
         return merged
 
-    @staticmethod
-    def _merge_images(item_a: DatasetItem, item_b: DatasetItem) -> Image:
-        image = None
+    @classmethod
+    def _merge_media(cls, item_a: DatasetItem, item_b: DatasetItem) \
+            -> Union[ByteImage, Image, PointCloud, Video, VideoFrame]:
+        if isinstance(item_a.media, (ByteImage, Image, VideoFrame)) or \
+                isinstance(item_b.media, (ByteImage, Image, VideoFrame)):
+            media = cls._merge_images(item_a, item_b)
+        else:
+            if isinstance(item_a.media, PointCloud) and isinstance(item_b.media, PointCloud) or \
+                    isinstance(item_a.media, MediaElement) and isinstance(item_b.media, MediaElement) or \
+                    isinstance(item_a.media, Video) and isinstance(item_b.media, Video):
+                if item_a.media.path and item_b.media.path and \
```

Comment:
Video needs to be compared for start and stop frames, Point cloud have to be merged on extra images

Reply:
Fixed. 

But I'm not sure that I understand correctly how to merge point cloud and video.

Reply:
I don't think we can merge video, because if there are different frame constraints, the videos are different. It will be reasonable to throw an error in this case.

For point clouds we can append extra images, if they are not specified, and we can append pcd path, if it's unknown.

Reply:
I think I fixed it